### PR TITLE
allow fileRoot and overrides to config to be injected allowing servin…

### PR DESCRIPTION
**Pull request by jamesgorrie**
…g assets from s3

Allowing us to serve assets from S3.
Behind a flag incase it's crap.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Couple of rudimentary response times:

from origin 2s => 0.2
from cache 0.2 => 0.1

I'm not sure why the cache is different, although S3 has eTags etc setup, so could be that.

File: server/views/layout/default.njk [hunk 1]
```diff
@@ -12,17 +12,17 @@
...
-    <link rel="icon" type="image/png" href="/assets/icons/favicon-16x16.png" sizes="16x16"/>
-    <link rel="manifest" href="/assets/icons/manifest.json"/>
-    <link rel="mask-icon" href="/assets/icons/safari-pinned-tab.svg" color="#000000"/>
-    <script src="/assets/libs/picturefill.min.js" async></script>
+    <link rel="shortcut icon" href="{{ '/assets/icons/' | getHashedFile('favicon.ico') }}" type="image/ico">
+    <link rel="icon" type="image/png" href="{{ '/assets/icons/' | getHashedFile('favicon-32x32.png') }}" sizes="32x32"/>
+    <link rel="icon" type="image/png" href="{{ '/assets/icons/' | getHashedFile('favicon-16x16.png') }}" sizes="16x16"/>
+    <link rel="manifest" href="{{ '/assets/icons/' | getHashedFile('manifest.json') }}'"/>
+    <link rel="mask-icon" href="{{ '/assets/icons/' | getHashedFile('safari-pinned-tab.svg') }}" color="#000000"/>
+    <script src="{{ '/assets/libs/' | getHashedFile('picturefill.min.js') }}" async></script>
```
Comment (jamesgorrie) [inline server/views/layout/default.njk]: Not sure I like this as a function.
@wellcometrust/experience-dev Any idea?

Comment (davidpmccormick) [reply server/views/layout/default.njk]: what are the other options? I think it looks OK.

Comment (jamesgorrie) [reply server/views/layout/default.njk]: I haven't thought of any, if you're good, I'm 😆

**Pull request merged**